test/jmap/email/search/use_case_test/search_by_after_use_case_test.dart

```diff
@@ -0,0 +1,185 @@
+                    "notFound": [],
+                    "state": "94bbff20-e87c-11ec-aae4-43ebf0340ebd",
+                    "list": [
+                      {
+                        "preview": "AAAA",
+                        "hasAttachment": false,
+                        "size": 3328,
+                        "subject": "AAAA",
+                        "from": [
+                          {
+                            "name": "Manh tuan Manh",
+                            "email": "manh199672@gmail.com"
+                          }
+                        ],
+                        "sentAt": "2022-06-10T04:44:03Z",
+                        "id": "04f27c50-e879-11ec-aae4-43ebf0340ebd",
+                        "receivedAt": "2022-06-10T04:51:41Z"
+                      }
+                    ]
+                  },
+                  "c2"
+                ]
+              ]
+            }),
+        data: {
+          "using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail"],
+          "methodCalls": [
+            [
+              "Email/query",
+              {
+                "accountId":
+                    "0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb",
+                "filter": {"after": "2022-05-12T06:12:00.000Z", "from": "manh"}
+              },
+              "c1"
+            ],
+            [
+              "Email/get",
+              {
+                "accountId":
+                    "0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb",
+                "#ids": {
+                  "resultOf": "c1",
+                  "name": "Email/query",
+                  "path": "ids/*"
+                },
+                "properties": [
+                  "id",
+                  "subject",
+                  "size",
+                  "from",
+                  "receivedAt",
+                  "sentAt",
+                  "preview",
+                  "hasAttachment"
+                ]
+              },
+              "c2"
+            ]
+          ]
+        },
+        headers: {
+          "accept": "application/json;jmapVersion=rfc-8621",
+          "content-type": "application/json; charset=utf-8",
+        });
+
+    final httpClient = HttpClient(dio);
+    final processingInvocation = ProcessingInvocation();
+    final jmapRequestBuilder =
+        JmapRequestBuilder(httpClient, processingInvocation);
+    final accountId = AccountId(
+      Id('0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb'),
+    );
+
+    final queryEmailMethod = QueryEmailMethod(accountId)
+      ..addFilters(
+        EmailFilterCondition(
+          after: UTCDate(
+            DateTime.parse('2022-05-12T06:12:00Z'),
+          ),
+          from: 'manh',
+        ),
+      );
+    final queryEmailInvocation = jmapRequestBuilder.invocation(queryEmailMethod,
+        methodCallId: MethodCallId('c1'));
```

Comment:
IMO there's no need to pass `methodCallId`, you should let it auto-generate. If I remember correctly, it starts with `c0`

Reply:
`methodCallId` is optional, `processingInvocation` handle to make it unique. But no problem if you want to use it yourself :)

Reply:
@dab246 I think if I add `methodCallId`, method request will be clearer